Rubberduck.RibbonDispatcher/Concrete/RibbonButton.cs

```diff
@@ -0,0 +1,69 @@
+﻿////////////////////////////////////////////////////////////////////////////////////////////////////
+//                                Copyright (c) 2017 Pieter Geerkens                              //
+////////////////////////////////////////////////////////////////////////////////////////////////////
+using System;
+using System.Resources;
+using System.Runtime.InteropServices;
+using System.Diagnostics.CodeAnalysis;
+using stdole;
+
+using Rubberduck.RibbonDispatcher.AbstractCOM;
+
+namespace Rubberduck.RibbonDispatcher.Concrete {
+    /// <summary>The ViewModel for Ribbon Button objects.</summary>
+    [Serializable]
+    [CLSCompliant(true)]
+    [ComVisible(true)]
+    [ClassInterface(ClassInterfaceType.None)]
+    [ComSourceInterfaces(typeof(IClickedEvents))]
+    [SuppressMessage("Microsoft.Interoperability", "CA1409:ComVisibleTypesShouldBeCreatable",
+        Justification = "Publc, Non-Creatable class with exported Events.")]
+    [ComDefaultInterface(typeof(IRibbonButton))]
+    [Guid(RubberduckGuid.RibbonButton)]
+    public class RibbonButton : RibbonCommon, IRibbonButton {
+        internal RibbonButton(string id, ResourceManager mgr, bool visible, bool enabled, RdControlSize size,
+                string imageMso, bool showImage, bool showLabel, EventHandler onClickedAction)
+            : base(id, mgr, visible, enabled, size){
+            _image     = new ImageObject(imageMso);
+            _showImage = showImage;
+            _showLabel = showLabel;
+            if (onClickedAction != null) Clicked += onClickedAction;
+        }
+        internal RibbonButton(string id, ResourceManager mgr, bool visible, bool enabled, RdControlSize size,
+                IPictureDisp image, bool showImage, bool showLabel, EventHandler onClickedAction)
+            : base(id, mgr, visible, enabled, size){
+            _image     = new ImageObject(image);
+            _showImage = showImage;
+            _showLabel = showLabel;
+            if (onClickedAction != null) Clicked += onClickedAction;
+        }
+
+        /// <inheritdoc/>
+        public event EventHandler Clicked;
+        /// <inheritdoc/>
+        public void OnAction() => Clicked?.Invoke(this, null);
+
+        #region IImageableItem implementation
```

Comment:
regions are a codesmell warranting extraction to a separate class that can be composed into the original class to maintain the same interface

Reply:
Agreed in general. I have made two attempts to clean this up, and it just gets worse instead of better. The ImageObject object already does most of the dirty work, and this seems to be the minimum code necessary for the composition. I am open to specific suggestions on how to implement this more cleanly, but I am out of ideas for this currently.

Reply:
@Vogel612 : The more I think on your comment, the deeper I realize that both question and answer are. I am essentially implementing the Decorator pattern here, except that it is on named classes instead of interfaces due to COM limitations. The point of the named regions is that these blocks of code MUST BE CHARACTER-BY-CHARACTER IDENTICAL across multiple classes - else the difference is a potential bug. So a ToggleButton is a RibbonCommon decorated with Imageable, Sizeable, and Toggleable, while a Checkbox is a RibbonCommon decorated only with Toggleable.